api/api_ethereum.go

```diff
@@ -1002,11 +1001,11 @@ func newEthTransactionReceipt(header *types.Header, tx *types.Transaction, b Bac
 		"type":              hexutil.Uint(byte(typeInt)),
 	}
 
-	if b.ChainConfig().IsEthTxTypeForkEnabled(new(big.Int).SetUint64(blockNumber)) {
-		fields["effectiveGasPrice"] = hexutil.Uint64(tx.EffectiveGasPrice(header.BaseFee).Uint64())
-	} else {
-		fields["effectiveGasPrice"] = hexutil.Uint64(tx.GasPrice().Uint64())
-	}
+	// After KIP-71 hard fork : return header.baseFee
+	// After EthTxType hard fork : use zero baseFee to calculate effective gas price for EthereumDynamicFeeTx.
+	//    return gas price of tx.
+	// Before EthTxType hard fork : return gas price of tx. (typed ethereum txs are not available.)
+	fields["effectiveGasPrice"] = hexutil.Uint64(tx.EffectiveGasPrice(header).Uint64())
```

Comment:
If we change code like this, then how can we get gas price of tx and put it into receipt field?

Reply:
`tx.EffectivePrice` function will return baseFee if KIP-71 hard fork is enabled.
And if KIP-71 hard fork is not enabled, return gas price of tx except for TxTypeEthereumDynamicFee (handling this tx type separate because there is no gas price field).

Reply:
Oh got it. Now cleared.